docs/channel_api.rst

```diff
@@ -0,0 +1,274 @@
+
+A channel may have many nodes using it, but tens not 1000s.
+
+The current reference implementation at TODO assumes that the node operator is also the channel endpoint operator, therefore manual devops style auth configuration is fine (eg. subnet only networking/whitelisting IP addresses/API Gateway SIG4 certs etc...).
+
+
+Channel Posting API
+-------------------
+
+| ``POST /messages``
+| ``GET /messages/<id>``
+| ``GET /messages/<id>?fields=status``
+
+.. uml::
+
+   @startuml
+   caption Posting a message to a channel
+   hide footbox
+   
+   box "Local Node" #LightGreen
+       participant Message_API
+       participant Channel_Posting_API
+   end box
+   participant Channel_Medium
+   box "Foreign Node" #LightBlue
+       participant Foreign_Node
+   end box
+   
+   Message_API->Channel_Posting_API: post message
+   activate Channel_Posting_API
+   return id
+
+   Channel_Posting_API->Channel_Medium: write message
+   alt subscribed to updates
+       Message_API->Channel_Posting_API: subscribe to updates
+       Channel_Posting_API->Message_API: <callback> update message status
+   else polls for updates
+       Message_API->Channel_Posting_API: <poll> get message status
+   end
+   Channel_Medium->Foreign_Node: receives message from channel
+   @enduml
+
+
+.. uml::
+
+   @startuml
+   hide empty description
+   caption State of a message posted to a Channel Posting API
+
+   [*] --> Received
+   Received -right-> Confirmed
+   Received --> Undeliverable
+   Confirmed --> Revoked
+   Revoked --> [*]
+   Undeliverable --> [*]
+   Confirmed -[dashed]-> [*]
+   @enduml
+
+
+States:
+
+ - **Received**: The message either hasn't been written to the channel (perhaps the first attempt errored and will be attempted again) or has been written but awaiting confirmation.
+ - **Confirmed**: The message has passed through the channel. Effectively the end state for most successful messages.
+
+   + On a blockchain, this means that there are sufficient blocks on top.
+   + On a DB this means that the message was commit to the table.
+
+ - **Undeliverable**: The channel was unable to write the message and has stopped trying
+ - **Revoked**: Confirmation was erroneously issued on a fork. We expect this to be extremely rare; it is a theoretical possibility.
+
+
+A typical BlockchainChannel:
+
+ - received message and writes to a DB, returning an ID
+ - writes to the blockchain
+ - waits (forever; stays in Received) and observes until:
+
+   + multiple blocks are written on top of the chain (Confirmed)
+   + OR observes that it was on a fork and the chain has moved from a previous block and the message was never written (Undeliverable)
+
+It is the channel API's business to decide if it fails as Undeliverable on the first attempt, or whether it tries a few times (config value) before being marked as Undeliverable.
+
+
+Channel Receiving API
```

Comment:
I don't understand this bit. If we are using websub, wouldn't the process be: 
1. the posting api advertises it's hub in response headers
2. the igl node posting to the channel api subscribes to the topic advertised in the response from the post the the channel api and posts a subscription to the advertised hub
3. the channel publishes events to it's hub
?

Reply:
This is internal to the posting api and talking about how it might try to put the message on the chain multiple times before deciding it's a failure.

Yes, I agree that that's how the node would subscribe to updates about the message it requested to be sent, and that's not documented there currently. We glossed over it initially for simplicity, but should work essentially the same as the channel receiving api's websub implementation.